x-pack/plugin/profiling/src/main/java/org/elasticsearch/xpack/profiling/TransportGetStackTracesAction.java

```diff
@@ -381,7 +491,56 @@ public void onResponse(MultiGetResponse multiGetItemResponses) {
                     }
                 }
             }
-            if (this.remainingSlices.decrementAndGet() == 0) {
+            mayFinish();
+        }
+
+        public void onHostsResponse(SearchResponse searchResponse) {
+            SearchHit[] hits = searchResponse.getHits().getHits();
+            for (SearchHit hit : hits) {
+                HostMetadata host = HostMetadata.fromSource(hit.getSourceAsMap());
+                hostsTable.putIfAbsent(host.hostID, host);
+            }
+            log.debug(hostsWatch::report);
+            log.debug("Have [{}] host metadata items", hostsTable.size());
+
+            calculateCO2AndCosts();
```

Comment:
As a consequence of my prior comment we'd delete the call here and move it to `mayFinish()`.